### PR TITLE
multi-collateral + feedbase + whitelists

### PR DESCRIPTION
**Pull request by nmushegian**

### CODE REVIEW

File: contracts/ss.sol [hunk 1]
```diff
@@ -1,152 +1,164 @@
...
+        purchased_quantity = (10**18 * pay_how_much) / (price + (price/t.spread));
         if(!safeToAdd(_balances[msg.sender], purchased_quantity))
             throw;
         if(!safeToAdd(_supply, purchased_quantity))
             throw;
         _balances[msg.sender] += purchased_quantity;
         _supply += purchased_quantity;
-        if( _supply > _max_supply ) throw;
-        return purchased_quantity;
+        t.current_debt += purchased_quantity;
```
Comment (rainbreak) [inline contracts/ss.sol:139]: needs check if safeToAdd, at least for consistency.

File: contracts/ss.sol [hunk 2]
```diff
@@ -1,152 +1,164 @@
...
+            throw;
+        }
+        var price = getPrice(t.feedID);
         if( _balances[msg.sender] < stablecoin_quantity )
             throw;
         _balances[msg.sender] -= stablecoin_quantity;
         _supply -= stablecoin_quantity;
-        returned_ether = (stablecoin_quantity * (_exchange_price-_exchange_spread)) / (10**18);
-        if( !msg.sender.call.value(returned_ether)() ) {
+        t.current_debt -= stablecoin_quantity;
```
Comment (rainbreak) [inline contracts/ss.sol:157]: Similarly I think this needs safeSub.

File: contracts/ss.sol [hunk 3]
```diff
@@ -1,152 +1,164 @@
...
-        purchased_quantity = 10**18 * msg.value / (_exchange_price+_exchange_spread);
+        var t = _types[collateral_type];
+        if( t.token == address(0) ) { // deleted
+            throw;
+        }
+        if( !t.token.transferFrom(msg.sender, t.vault, pay_how_much) ) {
+            throw;
+        }
+        var price = getPrice(t.feedID);
+        purchased_quantity = (10**18 * pay_how_much) / (price + (price/t.spread));
```
Comment (rainbreak) [inline contracts/ss.sol:132]: Do you still need to hardcode 10 *\* 18 now it is multi token?

File: contracts/ss.sol [hunk 4]
```diff
@@ -1,152 +1,164 @@
...
     {
-        _max_supply = max_supply;
+        _types[col_type].feedID = feed_id;
     }
-    function deposit()
-        ownerOnly
-    { /* Accept ether from owner */ }
 
-    function withdraw(uint quantity)
+    function registerCollateralType(ERC20 token, address vault, uint24 feedID, uint spread)
```
Comment (rainbreak) [inline contracts/ss.sol:98]: not ownerOnly? Allows malicious token injection.

or maybe whitelisted

**Comment by dbrock** [comment]
I'm going to delete all branches and keep only `master` now that `gh-pages` is no longer necessary https://github.com/blog/2233-publish-your-project-documentation-with-github-pages

**Pull request merged**